cmd/snap-bootstrap/cmd_initramfs_mounts.go

```diff
@@ -121,56 +143,75 @@ func copyUbuntuDataAuth(src, dst string) error {
 		"system-data/var/lib/extrausers/*",
 		"system-data/etc/ssh/*",
 		"user-data/*/.ssh/*",
+		// this ensures we get proper authentication to snapd from "snap"
+		// commands in recover mode
+		"user-data/*/.snap/auth.json",
 		// this ensures we also get non-ssh enabled accounts copied
 		"user-data/*/.profile",
 		// so that users have proper perms, i.e. console-conf added users are
 		// sudoers
 		"system-data/etc/sudoers.d/*",
+		// so that the time in recover mode moves forward to what it was in run
+		// mode
+		// NOTE: we don't sync back the time movement from recover mode to run
+		// mode currently, unclear how/when we could do this, but recover mode
+		// isn't meant to be long lasting and as such it's probably not a big
+		// problem to "lose" the time spent in recover mode
+		"system-data/var/lib/systemd/timesync/clock",
```

Comment:
not really related to the back port, but this probably should move on master to the other function renamed instead of being with the auth stuff

Reply:
Sure I can add a TODO for myself to do this with another related PR to master